lib/basedriver/capabilities.js

```diff
@@ -148,7 +148,7 @@ function parseCaps (caps, constraints = {}, shouldValidateCaps = true) {
   // Check for non-prefixed, non-standard capabilities and log warnings if they are found
   let nonPrefixedCaps = findNonPrefixedCaps(caps);
   if (!_.isEmpty(nonPrefixedCaps)) {
-    throw new errors.InvalidArgumentError(`All capabilities should have vendor prefix. The following capabilities should have vendor prefix: ${nonPrefixedCaps}`);
+    throw new errors.InvalidArgumentError(`All capabilities should have vendor prefix. The following capabilities did not have it: ${nonPrefixedCaps}`);
```

Comment:
should we rather throw session creation error?

Reply:
Yea, so far, https://github.com/appium/appium-base-driver/pull/440/files#diff-34fdfd47e5437d2bea9ab22b2fc976e52bb7d877d3be9dabdc597c1cf0d2c0c9R36 catches errors and it sends `SessionNotCreatedError` to the client.

This method itself did not createSession specific, so I handled this error as try/catch in createSession so that Appium can return SessionNotCreatedError in errors in createSession.